File: hazelnet-community/src/main/kotlin/io/hazelnet/community/data/discord/whitelists/WhitelistPartial.kt
package io.hazelnet.community.data.discord.whitelists

import com.fasterxml.jackson.annotation.JsonCreator
import io.hazelnet.community.data.discord.DiscordRequiredRole
import java.util.*
import javax.validation.Valid

data class WhitelistPartial @JsonCreator constructor(
    val displayName: String?,
    val maxUsers: Int?,
    val signupAfter: Date?,
    val signupUntil: Date?,
    val closed: Boolean?,
    val sharedWithServer: Int?,
    val launchDate: Date?,
    val logoUrl: String?,
    val awardedRole: Long?,
    @field:Valid
    val requiredRoles: MutableSet<DiscordRequiredRole>?,
)

File: vibrantnet/core/src/types/whitelist/whitelistPartial.ts
import { BlockchainType } from '../blockchainType';
import { DiscordRequiredRole } from '../discordRequiredRole';

export type WhitelistPartial = {
  displayName?: string | null
  maxUsers?: number | null
  signupAfter?: string | null
  signupUntil?: string | null
  closed?: boolean | null
  sharedWithServer?: number | null
  launchDate?: string | null
  logoUrl?: string | null
  awardedRole?: string | null
  requiredRoles?: DiscordRequiredRole[] | null
  blockchain?: BlockchainType[] | null
};
